### PR TITLE
ticket updates attribute/metric additions

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
This PR includes the following changes:

- The addition of ticket updates attributes and metrics highlighted within the [Zendesk Ticket Updates Dataset](https://support.zendesk.com/hc/en-us/articles/360022182114-Metrics-and-attributes-for-Zendesk-Support#topic_as3_slp_4y) documentation. Please note a high number of metrics were covered in the previous dbt_zendesk PR #12 
- Minor ReadMe and .yml documentation updates to reflect the appropriate model changes.
- Notable change to the `int_zendesk__field_pivot` model with down stream effects for incorporating the `user_id` of the updater.

To ensure all Zendesk Ticket Updates fields are included you may reference the working Zendesk Support Metrics Google Sheet with my notes on where the fields can be found within the package if not explicitly created in this PR.

## Joe's PR Overview Loom Video
[image](https://www.loom.com/share/d9e5c097028b41bca9cd37e13412a3f4)

### CODE REVIEW

File: models/intermediate/int_zendesk__ticket_historical_satisfaction.sql [hunk 1]
```diff
@@ -45,7 +56,23 @@ with satisfaction_updates as (
...
+
+    group by 1, 2, 3, 4, 5, 6
+
+), final as (
+    select
+        *,
+        case when lower(first_satisfaction_score) = 'good' and lower(latest_satisfaction_score) = 'bad'
+            then true
+            else false
+                end as is_good_to_bad_satisfaction_score,
```
Comment (fivetran-kristin) [inline models/intermediate/int_zendesk__ticket_historical_satisfaction.sql:70]: For is_good_to_bad_satisfaction_score and is_bad_to_good_satisfaction_score, you need to actually look at the last value, not necessary the first value. 

For example, suppose a ticket when from good -> bad -> good. The final change is considered a `is_bad_to_good_satisfaction_score`. The current logic won't take into consideration the second score.

File: models/zendesk.yml [hunk 2]
```diff
@@ -193,10 +193,16 @@ models:
         description: The unique key of the table, a surrogate key of date_day and ticket_id.
       - name: assignee_id
         description: The assignee id assigned to the ticket
+      - name: assignee_id_updater_id
```
Comment (fivetran-kristin) [inline models/zendesk.yml:196]: just updater_id?

**Review by fivetran-kristin — CHANGES_REQUESTED** [review]
One question added about the satisfaction ratings work, otherwise looks good.

**Comment by fivetran-kristin** [comment]
QQ - how difficult would it to include the updater_name rather than updater_Id in the zendesk__ticket_field_history table? I just ran it as an example and find it pretty hard to read with all the IDs

File: models/intermediate/int_zendesk__ticket_historical_satisfaction.sql (hunk 1 above)
Comment (fivetran-joemarkiewicz) [reply models/intermediate/int_zendesk__ticket_historical_satisfaction.sql:70]: Ah yeah I see where the logic won't take into consideration the second score in this case. I can update this.

**Comment by fivetran-joemarkiewicz** [comment]
I was able to fix the satisfaction scores good -> bad and bad -> good flag with some additional logic within the [int_zendesk__ticket_historical_satisfaction](https://github.com/fivetran/dbt_zendesk/blob/ticket-updates/models/intermediate/int_zendesk__ticket_historical_satisfaction.sql) model.

Additionally, I was able to leverage logic used within the dbt_jira package and create a new upstream [int_zendesk__field_history_enriched](https://github.com/fivetran/dbt_zendesk/blob/ticket-updates/models/ticket_history/int_zendesk__field_history_enriched.sql) model which joins the user information prior to pivoting. The `field_updater_name` was then easily added to the field history model for each tracked field. We can easily extrapolate this for the additional updater fields if we would like to remove the `field_updater_id`(s) altogether.

File: models/ticket_history/int_zendesk__field_history_enriched.sql [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+), users as (
+    select *
+    from {{ ref('stg_zendesk__user') }}
+
+), final as (
+    select
+        ticket_field_history.*,
+        case when ticket_field_history.user_id = -1
+            then 'zendesk_auto_change'
```
Comment (fivetran-joemarkiewicz) [inline models/ticket_history/int_zendesk__field_history_enriched.sql:14]: The `-1` fields are the Zendesk automated changes. Let me know if you would prefer a different name for these.

**Comment by fivetran-joemarkiewicz** [comment]
@fivetran-kristin I created a second Loom video going into high level some of the updates applied after your initial review. One thing I forgot to mention in the video is that all the appropriate updater fields are outlined within the ReadMe to allow users to simply copy and paste the relevant fields into their own `dbt_project.yml`.

Let me know if you have any questions!

## Joe's 2nd PR Overview Loom Video
[image](https://www.loom.com/share/23e443bc93394f478beec93fa8d2e986)

File: models/intermediate/int_zendesk__ticket_historical_satisfaction.sql [hunk 4]
```diff
@@ -42,10 +73,34 @@ with satisfaction_updates as (
     left join latest_comment
         using(ticket_id)
 
-    left join latest_score
+    left join first_and_latest_score
```
Comment (fivetran-kristin) [inline models/intermediate/int_zendesk__ticket_historical_satisfaction.sql:76]: can you use "using" here, given that multiple tables have ticket_id? 

I think this works fine in BQ, but not the other warehouses (though I may be wrong)

File: models/ticket_history/int_zendesk__field_history_enriched.sql [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+), updater_info as (
+    select *
+    from {{ ref('int_zendesk__updater_information') }}
+
+), final as (
+    select
+        ticket_field_history.*
+
+        {% if var('ticket_field_history_updater_columns')%} --The below will be run if any fields are included in the variable within the dbt_project.yml.
+            {% for col in var('ticket_field_history_updater_columns') %} --Iterating through the updater fields included in the varuable.
```
Comment (fivetran-kristin) [inline models/ticket_history/int_zendesk__field_history_enriched.sql:15]:
```suggestion
            {% for col in var('ticket_field_history_updater_columns') %} --Iterating through the updater fields included in the variable.
```

File: models/ticket_history/int_zendesk__field_history_pivot.sql [hunk 6]
```diff
@@ -1,3 +1,5 @@
+-- depends_on: {{ ref('stg_zendesk__ticket_field_history') }}
```
Comment (fivetran-kristin) [inline models/ticket_history/int_zendesk__field_history_pivot.sql:1]: do we need this here?

File: models/ticket_history/int_zendesk__field_history_enriched.sql [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+                --The below case when statements are needed to populate Zendesk automated fields for -1 user_id. Or, pass through the user defined -1 fields.
+                {% if col in ['updater_is_active'] %}
+                ,case when ticket_field_history.user_id = -1
```
Comment (fivetran-kristin) [inline models/ticket_history/int_zendesk__field_history_enriched.sql:19]: remove case whens, but leave if/elif logic for differing data types

File: models/intermediate/int_zendesk__ticket_historical_satisfaction.sql (hunk 4 above)
Comment (fivetran-joemarkiewicz) [reply models/intermediate/int_zendesk__ticket_historical_satisfaction.sql:76]: This does work fine in BQ and looks to run fine in RS and SF, but with our limited testing dataset I am not 100%. Removed "using" in joins just to be safe!

File: models/ticket_history/int_zendesk__field_history_enriched.sql (hunk 5 above)
Comment (fivetran-joemarkiewicz) [reply models/ticket_history/int_zendesk__field_history_enriched.sql:15]: Updated

File: models/ticket_history/int_zendesk__field_history_pivot.sql (hunk 6 above)
Comment (fivetran-joemarkiewicz) [reply models/ticket_history/int_zendesk__field_history_pivot.sql:1]: We need this here in order for the run_query statement to properly reference the ticket_field_history model

**Review by fivetran-kristin — APPROVED** [review]

**Pull request merged**